### ISSUE TITLE
ICE in solidity::u256 solidity::frontend::RationalNumberType::literalValue

### ISSUE THREAD

**Issue by agroce** [ethereum/solidity#8297]
## Description

This contract:
```solidity
contract c {
    function f() public pure {
        int a;
             a** 1E1233;
    }
}
```

produces:

```
Internal compiler error during compilation:
/home/user/solidity/libsolidity/ast/Types.cpp(1177): Throw in function virtual solidity::u256 solidity::frontend::RationalNumberType::literalValue(const solidity::frontend::Literal *) const
Dynamic exception type: boost::exception_detail::clone_impl<solidity::langutil::InternalCompilerError>
std::exception::what: Number constant too large.
[solidity::util::tag_comment*] = Number constant too large.
```

when run with solc

On master, using AFL fuzzing.  Another discovery using https://github.com/agroce/afl-compiler-fuzzer

## Environment

- Compiler version:  0.6.3-develop.2020.2.10+commit.e8eb1f2d.Linux.clang
- Target EVM version (as per compiler settings): N/A
- Framework/IDE (e.g. Truffle or Remix): N/A
- EVM execution environment / backend / blockchain client: N/A
- Operating system: Ubuntu 18.04 in docker

## Steps to Reproduce

Above shows pretty clearly, I think.

**Comment by chriseth** [comment]
Probably fixed by https://github.com/ethereum/solidity/issues/8291

**Comment by Marenz** [comment]
Unfortunately not fixed by #8291. Still happening in latest `develop`.
